### PR TITLE
Fix schema validation with array syntax for clone and services

### PR DESCRIPTION
**Pull request by lafriks**
Currently it gives warnings even when pipeline syntax is supported and compiles/runs just fine

### CODE REVIEW

**Review by xoxys — COMMENTED** [review]
LGTM in general.

Wasn't this removed/deprecated intentionally? I thought there were discussions somewhere in the past to not support both list and object-style syntax anymore, but I might be wrong

File: pipeline/frontend/yaml/linter/schema/.woodpecker/test-new-syntax.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+clone:
+  - name: git
+    image: woodpeckerci/plugin-git
+    settings:
+      partial: true
+  - name: testdata
+    image: woodpeckerci/plugin-git
+    settings:
+      remote: https://gitserver/owner/testdata.git
+      path: testdata
```
Comment (anbraten) [inline pipeline/frontend/yaml/linter/schema/.woodpecker/test-new-syntax.yml:12]: Wasn't aware that you could use this for multiple clone plugins as well

File: pipeline/frontend/yaml/linter/schema/schema_test.go [hunk 2]
```diff
@@ -101,6 +101,11 @@ func TestSchema(t *testing.T) {
 			testFile: ".woodpecker/test-broken.yml",
 			fail:     true,
 		},
+		{
+			name:     "New syntax",
+			testFile: ".woodpecker/test-new-syntax.yml",
+			fail:     false,
+		},
```
Comment (anbraten) [inline pipeline/frontend/yaml/linter/schema/schema_test.go:108]: I would suggest to call the file sth like `test-array-steps.yml` or `test-array-syntax.yml`.

**Review by anbraten — APPROVED** [review]

File: pipeline/frontend/yaml/linter/schema/schema_test.go [hunk 3]
```diff
@@ -101,6 +101,11 @@ func TestSchema(t *testing.T) {
 			testFile: ".woodpecker/test-broken.yml",
 			fail:     true,
 		},
+		{
+			name:     "New syntax",
```
Comment (anbraten) [inline pipeline/frontend/yaml/linter/schema/schema_test.go:105]:
```suggestion
			name:     "Array syntax",
```

**Review by xoxys — APPROVED** [review]

File: pipeline/frontend/yaml/linter/schema/.woodpecker/test-new-syntax.yml (hunk 1 above)
Comment (lafriks) [reply pipeline/frontend/yaml/linter/schema/.woodpecker/test-new-syntax.yml]: yes we are using this quite much, especially for test data, integration tests and sometimes in place of submodules and they also have some drawbacks

**Pull request merged**